src/NHibernate/Cfg/MappingSchema/HbmManyToMany.cs

```diff
@@ -83,14 +83,16 @@ public IEnumerable<object> ColumnsAndFormulas
 		{
 			get
 			{
-				if (Items != null)
-					return Items;
-				// Avoid a possible breaking change for mapping having left a column attribute along with a formula one.
-				// This is a mapping error, but previous implementation was silently ignoring the column attribute in
-				// such case.
-				if (!string.IsNullOrEmpty(formula))
-					return AsFormulas();
-				return AsColumns();
+				if (Items != null && (!string.IsNullOrEmpty(column) || !string.IsNullOrEmpty(formula)))
```

Comment:
Not sure if this is possible, but probably need to check `Items` for `null` or empty.

Reply:
It seems unlikely the xml deserialization will change its behavior, but after all some framework implementations may differ there. And mapping-by-code may also differ someday (it is not the case currently), so for robustness, it would be better to change this.